src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -431,7 +438,12 @@ class GroupReadsByUmi
 
   private val assigner = strategy.newStrategy(this.edits)
 
-  type ReadPair = (SamRecord, SamRecord)
+  /** Checks that the read's MAPQ is over a minimum, and if the read is paired, that the mate mapq is also over the min. */
+  private def mapqOk(rec: SamRecord, minMapQ: Int): Boolean = {
+    rec.mapq >= minMapQ &&
+      (!rec.paired || rec.get[Int](SAMTag.MQ.name()).exists(_ >= minMapQ))
```

Comment:
Why `.get` and `.exists` when earlier you said:
> the `MQ` tag is required on reads with mapped mates

Reply:
Good call.  I should add a check earlier on and fail if I have paired reads and MQ is absent, then I can just fetch it directly here.